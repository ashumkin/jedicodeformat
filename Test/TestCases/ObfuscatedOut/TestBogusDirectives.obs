unit testbogusdirectives;interface uses classes;var safecall,dispid:integer;cdecl
,protected:boolean;const readonly=10;type register=class(TObject)private fidynamic
,fipublic:integer;function getreadonly:boolean;protected public property dynamic:
integer read fidynamic write fidynamic;property public:integer read fipublic;property
readonly:boolean read getreadonly;function dispid:integer;safecall;function protected
:boolean;procedure contains;procedure requires;procedure package;function at:integer
;function on:integer;procedure final(buffer:tstrings);end;implementation label dynamic
;var platform:integer;procedure override;var virtual:integer;begin virtual:=10;end
;function register.dispid:integer;begin result:=12;end;function register.getreadonly
:boolean;var name:string;absolute:integer absolute name;begin name:='Fred!';result
:=true;end;function register.protected:boolean;begin result:=false;end;procedure 
Register.contains;var platform:array[1..4]of char;begin end;procedure Register.requires
;begin end;procedure Register.package;begin end;function register.at:integer;begin
result:=23;end;function register.on:integer;begin result:=23;end;procedure Register
.final(buffer:tstrings);begin end;procedure finalabuse(buffer:tstrings);begin with
Register.create do try final(buffer);finally free;end;end;end. 