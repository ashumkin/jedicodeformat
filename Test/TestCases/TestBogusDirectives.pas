unit TestBogusDirectives;

{ AFS 11 Jan 2000
 This unit compiles but is not semantically meaningfull
 it is test cases for the code formatting utility


 This unit has code that compiles, but is positively pathological,
 it mimics variable and proc names that are textualy identical to directives
 This is legal, but not advised
}

interface

uses Classes;

{ this also serves as a test of var layout }
var Safecall,
DispId: integer;
Cdecl, Protected: boolean;

const
ReadOnly = 10;

type
  Register = class (TObject)
    private
      fiDynamic, fiPublic: integer;
    function GetReadOnly: Boolean;
    protected
    public
      property Dynamic: integer rEad fiDynamic wRite fiDynamic;
      property Public: integer read fiPublic;
      property ReadOnly: Boolean rEAd GetReadOnly;

      function DispId: integer; SAFECAlL;
      function ProtecTED: boolean;

      { are these keywords or directives? }
      procedure Contains;
      procedure Requires;
      procedure Package;
      function At: integer;
      function On: integer;
      procedure Final(buffer: TStrings);
  end;

implementation

label dynamic;

var
  platform: integer;


procedure Override;
var
  Virtual: integer;
begin
  Virtual := 10;
end;

{ register }

function Register.Dispid: integer;
begin
  Result := 12;
end;

function Register.GetReadOnly: Boolean;
var
  Name: String;
  Absolute: integer    absolute            Name;
begin
  Name := 'Fred!';
  Result := True;
end;

function Register.ProtecTED: boolean;
begin
  Result := False;
end;

procedure Register.Contains;
var
  platform: array [1..4] of char;
begin
end;

procedure Register.Requires;
begin
end;

procedure Register.Package;
begin
end;

function Register.At: integer;
begin
  Result := 23;
end;

function Register.On: integer;
begin
  Result := 23;
end;

procedure Register.Final(buffer: TStrings);
begin
end;

procedure FinalAbuse(buffer: TStrings);
begin
  with Register.Create do
  try
    Final(buffer);
  finally
    Free;
  end;
end;


end.
