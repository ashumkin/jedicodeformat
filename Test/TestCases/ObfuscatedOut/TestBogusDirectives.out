unit testbogusdirectives;

interface

uses Classes;

var
  safecall, dispid: integer;
  cdecl, protected: boolean;

const
  ReadOnly = 10;

type
  Register = class(TObject)
  private
    fidynamic, fipublic: integer;
    function getreadonly: boolean;
  protected
  public
    property dynamic: integer Read fidynamic Write fidynamic;
    property public: integer Read fipublic;
    property ReadOnly: boolean Read getreadonly;
    function dispid: integer; safecall;
    function protected: boolean;
    procedure contains;
    procedure requires;
    procedure package;
    function at: integer;
    function on: integer;
    procedure final(buffer: TStrings);
  end;

implementation

label
  dynamic;

var
  platform: integer;

procedure override;
var
  virtual: integer;
begin
  virtual := 10;
end;

function Register.dispid: integer;
begin
  Result := 12;
end;

function Register.getreadonly: boolean;
var
  Name:     string;
  absolute: integer absolute Name;
begin
  Name   := 'Fred!';
  Result := True;
end;

function Register.protected: boolean;
begin
  Result := False;
end;

procedure Register.contains;
var
  platform: array[1..4] of char;
begin
end;

procedure Register.requires;
begin
end;

procedure Register.package;
begin
end;

function Register.at: integer;
begin
  Result := 23;
end;

function Register.on: integer;
begin
  Result := 23;
end;

procedure Register.final(buffer: TStrings);
begin
end;

procedure finalabuse(buffer: TStrings);
begin
  with
    Register.Create do
    try
      final(buffer);
    finally
      Free;
    end;
end;

end.
 